### ISSUE TITLE
Possibility to reset approvals count in gitlab when gitLabAutomerge set to true

### ISSUE THREAD

**Issue by SkReD** [renovatebot/renovate#5709]
**What would you like Renovate to be able to do?**
Gitlab allow to configure approvals count per repo, but dont have setting to auto-configure it per-mergre request depending on conditions. This is blocks automerge of merge request from renovate. 

**Describe the solution you'd like**
I want to have setting like `gitLabResetApprovers` possibly with `Automerge` postfix but it seems too long for setting. When set to true merge request would be created, approvals count set to zero and than automerge enabled.

**Describe alternatives you've considered**
Right now it possible to achieve with disabling automerge in renovate and adjust approvals count with subsequent automerge throught gitlab api but this is frustrating.

**Additional context**
Log part from pipeline
```
 DEBUG: Automerge on PR creation failed (repository=pfp/test-boxy-renovate, branch=renovate-noop)
        "err": {
          "name": "HTTPError",
          "hostname": "gitlab-rnd.tcsbank.ru",
          "method": "PUT",
          "path": "/api/v4/projects/pfp%2Ftest-boxy-renovate/merge_requests/11/merge",
          "protocol": "https:",
          "url": "https://gitlab-rnd.tcsbank.ru/api/v4/projects/pfp%2Ftest-boxy-renovate/merge_requests/11/merge",
          "gotOptions": {
            "path": "/api/v4/projects/pfp%2Ftest-boxy-renovate/merge_requests/11/merge",
            "protocol": "https:",
            "hostname": "gitlab-rnd.tcsbank.ru",
            "hash": "",
            "search": "",
            "pathname": "/api/v4/projects/pfp%2Ftest-boxy-renovate/merge_requests/11/merge",
            "href": "https://gitlab-rnd.tcsbank.ru/api/v4/projects/pfp%2Ftest-boxy-renovate/merge_requests/11/merge",
            "headers": {
              "user-agent": "https://github.com/renovatebot/renovate",
              "accept": "application/json",
              "accept-encoding": "gzip, deflate",
              "content-type": "application/json",
              "Private-token": "**redacted**",
              "content-length": 72
            },
            "hooks": {
              "beforeError": [],
              "init": [],
              "beforeRequest": [],
              "beforeRedirect": [],
              "beforeRetry": [],
              "afterResponse": []
            },
            "retry": {"methods": {}, "statusCodes": {}, "errorCodes": {}},
            "decompress": true,
            "throwHttpErrors": true,
            "followRedirect": true,
            "stream": false,
            "form": false,
            "json": true,
            "cache": false,
            "useElectronNet": false,
            "hostType": "gitlab",
            "baseUrl": "https://gitlab-rnd.tcsbank.ru/api/v4/",
            "body": "{\"should_remove_source_branch\":true,\"merge_when_pipeline_succeeds\":true}",
            "method": "PUT",
            "gotTimeout": {"request": 60000}
          },
          "statusCode": 406,
          "statusMessage": "Not Acceptable",
          "headers": {
            "server": "nginx",
            "date": "Fri, 13 Mar 2020 09:06:53 GMT",
            "content-type": "application/json",
            "content-length": "37",
            "connection": "close",
            "cache-control": "no-cache",
            "vary": "Origin",
            "x-content-type-options": "nosniff",
            "x-frame-options": "SAMEORIGIN",
            "x-request-id": "4ZRtHl87A81",
            "x-runtime": "0.072278"
          },
          "body": {"message": "Branch cannot be merged"},
          "message": "Response code 406 (Not Acceptable)",
          "stack": "HTTPError: Response code 406 (Not Acceptable)\n    at EventEmitter.<anonymous> (/usr/src/app/node_modules/got/source/as-promise.js:74:19)\n
```

**Comment by rarkins** [comment]
Is it really possible that a user or bot has permissions to "bypass" approvals for its own MRs?

**Comment by SkReD** [comment]
Yes. You can edit mr required approvals count even in merge request ui.

**Comment by viceice** [comment]
I don't think this feature is available in the opensource version. If I remember correctly it is a paid feature, named approval rules or something like that.

**Comment by SkReD** [comment]
It is configured by default approvals count in project settings, and than can be customized in concrete mr by manual editing. Just test it in cloud version at https://gitlab.com/SkReD/test/-/merge_requests/1. User with developer access can edit mr approvals count there.

**Comment by viceice** [comment]
![image](https://user-images.githubusercontent.com/1798109/76626702-f1a6c080-6539-11ea-9f3b-36e7452ad36d.png)
https://docs.gitlab.com/ee/api/merge_requests.html#approvals-starter

**Comment by SkReD** [comment]
ok. So if it is in paid version only, it should not be supported? And, anyway, in cloud it is supported on free plan

**Comment by viceice** [comment]
I don't say that we shouldn't support that scenario. I only want to mentions that this is a paid feature, see https://about.gitlab.com/pricing/

So it can be difficult to implement without a paid seat. (I prsonally don't have a paid seat)

Anyway I'm not sure, if this is a feature for the renovate bot. Maybe this should be done with another bot. like we use for our wn approvals on github: [renovate-approve](https://github.com/apps/renovate-approve)

**Comment by SkReD** [comment]
Cant find official way for creating such app for gitlab. 
I thought that adding such setting directly in renovate would be in line with `gitLabAutomerge` setting, that is already exists.

**Comment by rarkins** [comment]
Even if it's a feature for paid-only, I think it's fine to support. Our first decision is whether it belongs in Renovate or not. i.e. should Renovate have the option to bypass reviewers for its own MRs, or should there be "another bot". I'm thinking it's useful if Renovate can do it.

Next questions are how (which API endpoint or setting) and then finally naming.

**Comment by viceice** [comment]
ok, [here](https://docs.gitlab.com/ee/api/merge_request_approvals.html#merge-request-level-mr-approvals) is the api. Will be very complex, because the  rules seems to need to be check on several levels

**Comment by SkReD** [comment]
We use this one https://docs.gitlab.com/ee/api/merge_request_approvals.html#change-approval-configuration in our pipelines. What other levels beside mr itself shoud be checked?

**Comment by viceice** [comment]
this api is deprecated and will be remove with v13 (coming in may 2020?)
![image](https://user-images.githubusercontent.com/1798109/76642896-57a04180-6554-11ea-816e-d95a8d187df5.png)

**Comment by vegerot** [comment]
Is this problem related to https://github.com/renovatebot/config-help/issues/936#issuecomment-714681076 ?  It would be nice to have renovate auto-merge its MRs

**Comment by vegerot** [comment]
I've added the custom approval rule 
<img width="1101" alt="image" src="https://user-images.githubusercontent.com/5297725/96924380-9306d500-1480-11eb-919c-1ff68e92b64b.png">

So it seems to be getting us closer...
<img width="1079" alt="image" src="https://user-images.githubusercontent.com/5297725/96924450-a9149580-1480-11eb-89d8-85300d4076c8.png">

**Comment by viceice** [comment]
I think you are also interested in https://github.com/renovatebot/renovate/issues/6577

**Comment by rarkins** [comment]
The approvals requirement reminds me of this:
![image](https://user-images.githubusercontent.com/6311784/96958984-431b2300-14ff-11eb-8375-a8b75480418a.png)

**Comment by vegerot** [comment]
haha 😆 , how can I get around this two-foot wide fence?!!!

**Comment by HonkingGoose** [comment]
> Even if it's a feature for paid-only, I think it's fine to support. Our first decision is whether it belongs in Renovate or not. i.e. should Renovate have the option to bypass reviewers for its own MRs, or should there be "another bot". I'm thinking it's useful if Renovate can do it.
> 
> Next questions are how (which API endpoint or setting) and then finally naming.

@rarkins Did you decide whether this belongs in Renovate's repository or somewhere else? If you still want this issue, we still need to think about these two things at least:

1. How to implement behavior (which API endpoint or setting.
2. Naming the new feature.

**Comment by rarkins** [comment]
I think it's ok functionality but needs someone to work out the API flow necessary (we want non-deprecated endpoints to use)

**Comment by niklasweimann** [comment]
Hey 👋 
I would love to see this feature implemented, since we only execute all tests on merge requests. It seems there is no way to keep approval rules for developers while allowing renovate to automerge its merge requests, right?
